src/handlers/transfers/handler.js

```diff
@@ -177,21 +187,44 @@ const prepare = async (error, messages) => {
       throw fspiopError
     } else { // !hasDuplicateId
       const { validationPassed, reasons } = await Validator.validatePrepare(payload, headers)
-
-      if (Config.INCLUDE_DECODED_TRANSACTION_OBJECT) {
-        try {
-          const transactionObject = (new Ilp({ secret: null })).getTransactionObject(payload.ilpPacket)
-          message.value.content.transaction = transactionObject
-        } catch {
-          Logger.info('Ilp failed to decode. Error notification will be handled by validation check.')
+      // Select settlement model here
+      let settlementModel
+      try {
+        const allSettlementModels = await SettlementModelCached.getAll()
+        const settlementModels = allSettlementModels.filter(model => !model.currencyId || model.currencyId === payload.amount.currency)
+        if (settlementModels.length === 0) {
+          throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.GENERIC_SETTLEMENT_ERROR, 'Unable to find a matching or default, Settlement Model')
         }
+        settlementModel = settlementModels.find(sm => sm.ledgerAccountTypeId === Enum.Accounts.LedgerAccountType.POSITION)
+        if (Config.INCLUDE_DECODED_TRANSACTION_OBJECT) {
```

Comment:
I don't necessarily think we need this config option anymore.

Reply:
Done